Adding PR template to standardizing contributions (#1361)
intel/openfl: .github/pull_request_template.md

## Summary

> [Provide a short and descriptive title for the pull request, e.g., "Improve caching mechanism for API calls"]

Reviewer: Suggest rewording this section since the title will already be provided when opening the PR

```suggestion
[Provide a short and descriptive summary of the pull request, e.g., "Improve caching mechanism for API calls"]
```

Author: changes done in 9392476bd884757bacca50d4940066f8578e1a5f

## Type of Change (Mandatory)

> Specify the type of change being made. 

Reviewer 2: Could you provide a list of possible types of changes, for illustration purpose?

Author: changes done in 9392476bd884757bacca50d4940066f8578e1a5f

## Description (Mandatory)
[Explain the goal of this PR, the problem it solves and a summary of changes made. Reference any related issues or tickets (e.g., `Fixes #123`).]

## Testing
[Describe the testing done for this PR. If applicable include screenshots.]

<!-- ## Additional Information

> [Any additional information that reviewers should be aware of.] -->ßßß

Reviewer 3: Extra characters at the end?
```
ßßß
```

Author: resolved in 63b0015dccba96bef6d87dc7422a7a36fdc7044d

> # Pull Request Template

Reviewer 3: This can be removed from the template otherwise it would be the starting for each PR.

Author: changes made in 1fa03df4e4031e10344b3cb3e226311698b682b2

> ## PR Title

Reviewer 3: Would it not be already covered by the title which is a separate field when creating a PR?

Reviewer 4: How about "Summary"?

Author: changes made in 1fa03df4e4031e10344b3cb3e226311698b682b2

[Provide a short and descriptive title for the pull request, e.g., "Improve caching mechanism for API calls"]

> ## Type of Change (Mandatory)

Reviewer 4: This is a good improvement on the first draft.

My suggestion would be to remove this checkbox section. It helps in two ways:
1. Makes the template easier to approach (less overwhelming) and fits in a single screen. 
2. Description generally obviates the need for "Type of change" selection.

Author: changes made in 1fa03df4e4031e10344b3cb3e226311698b682b2

Specify the type of change being made. Use one of the following:
- [ ] New feature
- [ ] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] Test Addition
- [ ] Other (please specify)

## Description (Mandatory)

> [Provide a brief description of the changes made to this PR. Explain the goal of the PR, the problem it solves and main changes made in this PR. Reference any related issues or tickets (e.g., `Fixes #123`).]

Reviewer 4: ```suggestion
[Explain the goal of this PR, the problem it solves and a summary of changes made. Reference any related issues or tickets (e.g., `Fixes #123`).]
```

Author: changes made in 1fa03df4e4031e10344b3cb3e226311698b682b2

> ## Testing & Verification 

Reviewer 4: "Testing" is sufficient?

Author: changes made in 1fa03df4e4031e10344b3cb3e226311698b682b2

[Describe the testing done for this PR. If applicable include screenshots.]

> ## Additional Information

Reviewer 4: I suggest commenting this section. When a new PR is created, additional info section is visible, however not visible on the rendered markdown when this section is left blank (which is more often than not)

Author: changes made in 1fa03df4e4031e10344b3cb3e226311698b682b2

# Pull Request Template

> ## Title

Reviewer 5: Is this section supposed to be a part of the PR description or meant for PR title/subject only?

Author: resolved in  cefb8a50aa781e4653c4acc261617f08b1510e64

[Provide a short and descriptive title for the pull request, e.g., "Improve caching mechanism for API calls"]

## Type of Change (Mandatory)
Specify the type of change being made. Use one of the following:
- [ ] New feature
- [ ] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] Test Addition
- [ ] Other (please specify)

## Description (Mandatory)
[Provide a detailed explanation of the changes made in this PR. Explain the purpose of the PR and the problem it solves. Reference any related issues or tickets (e.g., `Fixes #123`).]

> ## Changes Made (Mandatory)

Reviewer 5: Just a suggestion - Description and Changed Made can be combined under single header to shorten the checklist.

Author: resolved in  cefb8a50aa781e4653c4acc261617f08b1510e64

List the key changes made in this PR. Be specific about files, functions, or modules that were modified. If applicable, include before-and-after code snippets or examples.

## Testing & Verification (Mandatory)
[Describe how the changes were tested. Include details about unit tests, integration tests, or manual testing. Specify any new tests added to cover the changes.]
- [ ] Unit tests added

> - [ ] Integration tests

Reviewer 5: I don't think we have "Integration Tests" in place, so this can be removed to avoid confusion.

Author: resolved in  cefb8a50aa781e4653c4acc261617f08b1510e64

- [ ] Manually tested

> - [ ] End-to-end (E2E) tests

Reviewer 5: Are we expecting successful GitHub workflow links from the users, if this option is selected ?

Author: resolved in cefb8a50aa781e4653c4acc261617f08b1510e64

- [ ] Other (please specify)

## Impact
[Discuss the impact of the changes on the project. This might include effects on performance, new dependencies, or changes in behaviour.]

> ## Documentation

Reviewer 3: This section can be removed as there is a documentation option in `Type of Change` section.
- The link can be provided in testing details. 
- Description goes in description section. 
- Screenshots can be added in the respective section as well.

Author: resolved in  cefb8a50aa781e4653c4acc261617f08b1510e64

Indicate if documentation was updated. Provide links to updated documentation or describe the changes made.

## Additional Information
[Any additional information that reviewers should be aware of.]

## Checklist (Mandatory)
Ensure the following tasks are completed before submitting the PR:
- [ ] Code follows the project's coding standards.
- [ ] All tests pass successfully.

> - [ ] Dependencies updated in `requirements.txt` or `pyproject.toml` (if applicable).

Reviewer 6: Suggest: group mandatory and optional checklist items, possibly create separate mandatory and optional checklist

Author: resolved in  cefb8a50aa781e4653c4acc261617f08b1510e64
